sstables/storage.cc

```diff
@@ -114,6 +117,38 @@ future<> filesystem_storage::rename_new_file(const sstable& sst, sstring from_na
     });
 }
 
+future<> filesystem_storage::sync_cached_directory(std::string_view name) const {
+    std::string key { name };
+    auto iter { _dircache.find(key) };
+    file dir_f;
+
+    if (iter == _dircache.end()) {
+        dir_f = co_await open_directory(name);
+
+        try {
+            auto res { _dircache.emplace(std::move(key), std::move(dir_f)) };
```

Comment:
could use structured binding for better readability.

Reply:
I agree. I vaguely recalled that modern C++ had something like that, but I couldn't remember the exact term, and so my quick web search came up empty as well. Will update.